pkgs/development/libraries/openexr/default.nix

```diff
@@ -9,49 +9,40 @@
 , libtool
 }:
 
-let
-  non_glibc_fpstate_patch =
-    # Fix ilmbase/openexr using glibc-only fpstate.
-    # Found via https://git.alpinelinux.org/aports/tree/community/openexr/10-musl-_fpstate.patch?id=80d9611b7b8e406a554c6f511137e03ff26acbae,
-    # TODO Remove when https://github.com/AcademySoftwareFoundation/openexr/pull/798 is merged and available.
-    #      Remove it from `ilmbase` as well then.
-    (fetchpatch {
-      name = "ilmbase-musl-_fpstate.patch.patch";
-      url = "https://raw.githubusercontent.com/void-linux/void-packages/80bbc168faa25448bd3399f4df331b836e74b85c/srcpkgs/ilmbase/patches/musl-_fpstate.patch";
-      sha256 = "0appzbs9pd6dia5pzxmrs9ww35shlxi329ks6lchwzw4f2a81arz";
-    });
-in
-
 stdenv.mkDerivation rec {
   pname = "openexr";
-  version = "2.4.1";
+  version = "2.5.2";
+
+  outputs = [ "bin" "dev" "out" "doc" ];
 
   src = fetchFromGitHub {
     owner = "AcademySoftwareFoundation";
     repo = "openexr";
     rev = "v${version}";
-    sha256 = "020gyl8zv83ag6gbcchmqiyx9rh2jca7j8n52zx1gk4rck7kwc01";
+    sha256 = "dtVoXA3JdmNs1iqu7cZlAdxt/CAgL5lSbOwu0SheyO0=";
   };
 
-  outputs = [ "bin" "dev" "out" "doc" ];
+  patches = [
+    # Fix pkg-config paths
+    (fetchpatch {
+      url = "https://github.com/AcademySoftwareFoundation/openexr/commit/6442fb71a86c09fb0a8118b6dbd93bcec4883a3c.patch";
+      sha256 = "bwD5WTKPT4DjOJDnPXIvT5hJJkH0b71Vo7qupWO9nPA=";
+    })
+  ] ++ lib.optionals (stdenv.hostPlatform.isLinux && stdenv.hostPlatform.libc != "glibc") [
+    # Fix ilmbase/openexr using glibc-only fpstate.
+    # Found via https://git.alpinelinux.org/aports/tree/community/openexr/10-musl-_fpstate.patch?id=80d9611b7b8e406a554c6f511137e03ff26acbae,
+    # TODO Remove when https://github.com/AcademySoftwareFoundation/openexr/pull/798 is merged and available.
+    (fetchpatch {
+      name = "ilmbase-musl-_fpstate.patch.patch";
+      url = "https://raw.githubusercontent.com/void-linux/void-packages/80bbc168faa25448bd3399f4df331b836e74b85c/srcpkgs/ilmbase/patches/musl-_fpstate.patch";
+      sha256 = "0appzbs9pd6dia5pzxmrs9ww35shlxi329ks6lchwzw4f2a81arz";
+    })
+  ];
   nativeBuildInputs = [ cmake libtool ];
   propagatedBuildInputs = [ ilmbase zlib ];
 
-  postPatch =
-    if (stdenv.hostPlatform.isLinux && stdenv.hostPlatform.libc != "glibc")
-      then
-        ''
-          patch -p0 < ${non_glibc_fpstate_patch}
```

Comment:
I don't think this can be removed, because the patch looks like

```
--- IlmBase/IexMath/IexMathFpu.cpp.orig
+++ IlmBase/IexMath/IexMathFpu.cpp
```

not starting with `a/IlmBase`, so `-p0` is required, and `patches` uses `-p1`.

Reply:
This fixes it:

```diff
-      sha256 = "0appzbs9pd6dia5pzxmrs9ww35shlxi329ks6lchwzw4f2a81arz";
+      sha256 = "1bmyg4qfbz2p5iflrakbj8jzs85s1cf4cpfyclycnnqqi45j8m8d";
+      # The patch's files are written as `IlmBase/...`, this turns it into
+      # `a/IlmBase/...`, so that the `patch -p1` that `patches` does works.
+      extraPrefix = ""; # Changing this requires changing the `sha256` (fixed-output)!
```

It works because of https://github.com/NixOS/nixpkgs/blob/07f111eac53bd351dc8e509751e1a2f5d38111ae/pkgs/build-support/fetchpatch/default.nix#L31-L34

which I find confusing; you'd think `extraPrefix = ""` doesn't do anything, but it puts an `a/` prefix in.

But I'll force-push that fix into this PR.